src/BuiltInTools/DotNetDeltaApplier/HotReloadAgent.cs

```diff
@@ -82,24 +82,36 @@ private UpdateHandlerActions GetMetadataUpdateHandlerActions()
             var handlerActions = new UpdateHandlerActions();
             foreach (var assembly in sortedAssemblies)
             {
-                foreach (var attr in assembly.GetCustomAttributesData())
+                try
                 {
-                    // Look up the attribute by name rather than by type. This would allow netstandard targeting libraries to
-                    // define their own copy without having to cross-compile.
-                    if (attr.AttributeType.FullName != "System.Reflection.Metadata.MetadataUpdateHandlerAttribute")
+                    foreach (var attr in assembly.GetCustomAttributesData())
                     {
-                        continue;
+                        // Look up the attribute by name rather than by type. This would allow netstandard targeting libraries to
+                        // define their own copy without having to cross-compile.
+                        if (attr.AttributeType.FullName != "System.Reflection.Metadata.MetadataUpdateHandlerAttribute")
+                        {
+                            continue;
+                        }
+
+                        IList<CustomAttributeTypedArgument> ctorArgs = attr.ConstructorArguments;
+                        if (ctorArgs.Count != 1 ||
+                            ctorArgs[0].Value is not Type handlerType)
```

Comment:
Nit: you can use array expression pattern: `if (ctorArgs is not [{Value: Type handlerType])`.

Reply:
This was originally written this way. The indentation change makes it show as a change, but it's not really.